engine/src/test/java/org/camunda/bpm/engine/test/api/mgmt/license/LicenseKeyTest.java

```diff
@@ -0,0 +1,164 @@
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.test.api.mgmt.license;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Map;
+
+import org.camunda.bpm.engine.ManagementService;
+import org.camunda.bpm.engine.ProcessEngine;
+import org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.camunda.bpm.engine.impl.cmd.LicenseCmd;
+import org.camunda.bpm.engine.test.api.resources.GetByteArrayCommand;
+import org.camunda.bpm.engine.test.util.ProcessEngineTestRule;
+import org.camunda.bpm.engine.test.util.ProvidedProcessEngineRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+public class LicenseKeyTest {
+
+  public ProvidedProcessEngineRule engineRule = new ProvidedProcessEngineRule();
+  public ProcessEngineTestRule testRule = new ProcessEngineTestRule(engineRule);
+
+  @Rule
+  public RuleChain ruleChain = RuleChain.outerRule(testRule).around(engineRule);
+
+  ProcessEngine processEngine;
+  ProcessEngineConfigurationImpl processEngineConfiguration;
+  ManagementService managementService;
+
+  @Before
+  public void init() {
+    processEngine = engineRule.getProcessEngine();
+    processEngineConfiguration = (ProcessEngineConfigurationImpl) processEngine.getProcessEngineConfiguration();
+    managementService = processEngine.getManagementService();
+  }
+
+  @After
+  public void tearDown() {
+    managementService.deleteLicenseKey();
+  }
+
+  @Test
+  public void testSetLicenseKey() {
+    // given
+    String licenseKey = "testLicenseKey";
+
+    // when
+    managementService.setLicenseKey(licenseKey);
+    Map<String, String> properties = managementService.getProperties();
+    String licenseKeyLegacyProperty = properties.get(LicenseCmd.LICENSE_KEY_PROPERTY_NAME);
+    String licenseByteArrayId = properties.get(LicenseCmd.LICENSE_KEY_BYTE_ARRAY_ID);
+
+    // then
+    assertThat(licenseKeyLegacyProperty).isNull();
+    assertThat(licenseByteArrayId).isNotNull();
+    // make sure a newly set license is not stored in properties...
+    assertThat(licenseByteArrayId).isNotEqualTo(licenseKey);
+    // ...but in the byte array table, referenced by the property
+    assertThat(processEngineConfiguration.getCommandExecutorTxRequired().execute(new GetByteArrayCommand(licenseByteArrayId)).getBytes())
+        .isEqualTo(licenseKey.getBytes());
+  }
+
+  @Test
+  public void testSetDuplicateLicenseKey() {
```

Comment:
1. The title is a bit misleading since we're not setting a duplicate here but simply a new license key, so we're rather testing that updating the key works, which is fine.
1. Adding a test for setting a duplicate would be a good idea.
1. We should test that setting a `null` license key works as expected.

Reply:
I agree